crates/common/trie/trie_iter.rs

```diff
@@ -138,9 +139,9 @@ impl Iterator for TrieIterator {
             return None;
         };
         // Fetch the last node in the stack
-        let (mut path, next_node_ref) = self.stack.pop()?;
+        let (mut path, mut next_node_ref) = self.stack.pop()?;
         let next_node = next_node_ref
-            .get_node(self.db.as_ref(), path.clone())
+            .get_node_mut(self.db.as_ref(), path.clone())
```

Comment:
This, for example, could be made into a `get_node_ref` call, since it doesn't need the mutability.